projects/js-packages/charts/src/hooks/use-chart-mouse-handler.ts

```diff
@@ -0,0 +1,89 @@
+import { localPoint } from '@visx/event';
+import { useTooltip } from '@visx/tooltip';
+import { useCallback, type MouseEvent } from 'react';
+import type { DataPoint } from '../components/shared/types';
+
+type UseChartMouseHandlerProps = {
+	/**
+	 * Whether tooltips are enabled
+	 */
+	withTooltips: boolean;
+};
+
+type UseChartMouseHandlerReturn = {
+	/**
+	 * Handler for mouse move events
+	 */
+	onMouseMove: ( event: React.MouseEvent< SVGElement >, data: DataPoint ) => void;
+	/**
+	 * Handler for mouse leave events
+	 */
+	onMouseLeave: () => void;
+	/**
+	 * Whether the tooltip is currently open
+	 */
+	tooltipOpen: boolean;
+	/**
+	 * The current tooltip data
+	 */
+	tooltipData: DataPoint | null;
+	/**
+	 * The current tooltip left position
+	 */
+	tooltipLeft: number | undefined;
+	/**
+	 * The current tooltip top position
+	 */
+	tooltipTop: number | undefined;
+};
+
+/**
+ * Hook to handle mouse interactions for chart components
+ *
+ * @param {UseChartMouseHandlerProps} props - Hook configuration
+ * @return {UseChartMouseHandlerReturn} Object containing handlers and tooltip state
+ */
+const useChartMouseHandler = ( {
+	withTooltips,
+}: UseChartMouseHandlerProps ): UseChartMouseHandlerReturn => {
+	const { tooltipOpen, tooltipLeft, tooltipTop, tooltipData, hideTooltip, showTooltip } =
+		useTooltip< DataPoint >();
+
+	const onMouseMove = useCallback(
```

Comment:
Do we need to debounce the calls here?

Reply:
That's a good point. I added a comment to address it and it's also tracked here https://github.com/Automattic/jetpack-roadmap/issues/2065